common/achievement/template.ts

```diff
@@ -88,4 +93,165 @@ async function getTemplatesByMetrics(metricsForAction: Metric[]) {
+    const group = await getTemplateGroup(template.group);
+    logger.info(`Checking AchievementTemplate(${template.id}) for consistency`, { template, group });
+
+    // ---- Template Metadata -----
+    if (!template.name || !template.description || !template.subtitle || !template.image) {
+        throw new PrerequisiteError(`AchievementTemplates need a name, description, subtitle and image`);
+    }
+
+    const actionFields = [!!template.actionName, !!template.actionRedirectLink, !!template.actionType];
+    if (actionFields.some((it) => it) && !actionFields.every((it) => it)) {
+        throw new PrerequisiteError(`actionName, actionRedirectLink and actionType must either all be set or all empty`);
+    }
+
+    // ---- Template Logic --------
+    const aggregations = Object.keys(template.conditionDataAggregations);
+    if (aggregations.length < 1) {
+        throw new PrerequisiteError(`AchievementTemplate needs at least one data aggregation`);
+    }
+
+    for (const [name, aggregation] of Object.entries(template.conditionDataAggregations as ConditionDataAggregations)) {
+        if (!isMetric(aggregation.metric)) {
+            throw new PrerequisiteError(`Aggregation ${name} uses unknown metric ${aggregation.metric}`);
+        }
+
+        // TODO: How to evaluate the aggregators?
+    }
+
+    // TODO: Does the template.type imply any prerequisities?
+
+    const sampleResult = await swan.parse(template.condition)(Object.fromEntries(aggregations.map((it) => [it, 0])));
+    if (typeof sampleResult !== 'boolean') {
+        throw new PrerequisiteError(
+            `AchievementTemplate condition does not evaluate to a boolean - This could be as it references to non existent aggregations`
+        );
+    }
+
+    // ---- Template Group --------
+    // We generally assume that if one template is activated that the whole group is supposed to be activated
+    // Thus also inconsistencies of not yet enabled achievement templates might show up here (this is easier to check)
+
+    // Check that groupOrders are sequential without gaps and that they are activated in sequence
+    let currentOrder = 1;
+    for (const groupTemplate of group) {
+        if (groupTemplate.groupOrder !== currentOrder) {
+            throw new PrerequisiteError(`Inconsistency in groupOrder, must be sequential`);
+        }
+
+        if (currentOrder < template.groupOrder && !groupTemplate.isActive) {
+            throw new PrerequisiteError(`Templates of a sequence must be activated in order`);
+        }
+
+        currentOrder += 1;
+    }
+
+    // Check that stepNames are set for a group with multiple steps
+    if (group.length > 1) {
+        for (const groupTemplate of group) {
+            if (!groupTemplate.stepName) {
+                throw new PrerequisiteError(`For templates of a sequence, every group template must have a stepName`);
```

Comment:
Is this correct? At least in the integration tests this is not followed?

Reply:
I think it's only correct for sequential achievements. Tiered are in a group as well, but don't have a step name.